### PR TITLE
Use M1 resource class for macos builds on CircleCI

### PR DESCRIPTION
**Pull request by arcondello**
I'll uncap the NumPy version in another PR (i.e., https://github.com/dwavesystems/dimod/pull/1373).

See also https://github.com/dwavesystems/dwave-ocean-sdk/issues/294.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -259,7 +267,7 @@ jobs:
             python -m venv env
             . env/bin/activate
             pip install dimod --no-index -f dist/ --force-reinstall --no-deps
-            pip install << parameters.dependencies >> --upgrade --only-binary=numpy
+            pip install "<< parameters.dependencies >>" --upgrade --only-binary=numpy
```
Comment (randomir) [inline .circleci/config.yml:270]: I recommend quoting individual specifiers -- below in the list of parameter values -- not the parameter value here. That way you can use multiple specifiers per parameter (e.g. `"'numpy<2' scipy~=1.0"`). For example, see cloud-client's config, or [dwave-inspector's](https://github.com/dwavesystems/dwave-inspector/blob/master/.circleci/config.yml) one.

Alternatively, you can use tilde in version specifier, and avoid quoting altogether.

File: .circleci/config.yml [hunk 2]
```diff
@@ -3,6 +3,7 @@ version: 2.1
 orbs:
   codecov: codecov/codecov@3
   win: circleci/windows@5.0.0
+  macos: circleci/macos@2.4.1
```
Comment (randomir) [inline .circleci/config.yml:6]: It's probably fine to reduce precision to minor or major version here, that way the latest is used.

**Review by randomir — APPROVED** [review]
lgtm

File: .circleci/config.yml (hunk 1 above)
Comment (arcondello) [reply .circleci/config.yml:270]: I will likely revert this commit as part of the NumPy 2.0 PR, so I will leave it be for now.

**Pull request merged**